### DISCUSSION TITLE
cam = torch.nansum(weight * activation.squeeze(0), dim=0)  # type: ignore[union-attr]

### DISCUSSION THREAD

**Discussion by hit-dyl** [frgfm/torch-cam#148] [Q&A] [answered]
```
cam = torch.nansum(weight * activation.squeeze(0), dim=0) # type: ignore[union-attr]
AttributeError: module 'torch' has no attribute 'nansum'
```
The function is missing

**Comment by frgfm** [comment, marked as answer]
Hi @hit-dyl :wave: 

Thanks for reporting this! However this looks more like a bug report (there is a Github issue template for this), please try to use those when appropriate :pray: The template walks you through how to provide required information for me to help you debug this :sweat_smile: 

That being said, seeing the error that you get, the problem doesn't come from this project directly. Which version of PyTorch are you using?

Your error shows that you are using an older version of PyTorch, as `torch.nansum` appeared in release `1.7.0`. If that's the case, simply upgrade your pytorch version and you'll be good to go :) 
On my end, I'll update the torch requirement on TorchCAM, because the current one is `>=1.5.1` while, as explained above, this now needs to be `>=1.7.0`. I hope this helps!

Cheers :v:

**Reply by hit-dyl** [reply]
Thank you. I see
